lib/fork.js

```diff
@@ -56,7 +56,7 @@ module.exports = (file, opts, execArgv) => {
 	const subprocess = childProcess.fork(workerPath, args, {
 		cwd: opts.projectDir,
 		silent: true,
-		env,
+		env: {...opts.environmentVariables, ...env},
```

Comment:
The `environmentVariables` option should override the env. So this needs to be reversed. This should also be documented in the readme.

Reply:
That was my initial thought. Then I realized that the `env` variable also contains `process.env` from the main process which makes it impossible to overwrite the configuration from the command line.

Reply:
Let's look at how `env` is built up: https://github.com/avajs/ava/blob/f26634b1a82f1edc92d75935c833c6a8cad7cb93/lib/fork.js#L12:L24

* We set a default `NODE_ENV` variable, though it is overridden by `process.env`
* We rewrite `NODE_PATH`
* We set `AVA_PATH`

I don't think `AVA_PATH` and `NODE_PATH` should be overridden by `environmentVariables`. But `NODE_ENV` should. All that said I think `environmentVariables` should override `process.env`. Which means we need to break this down further:

```
{NODE_ENV: 'test', ...process.env, opts.environmentVariables, NODE_PATH, AVA_PATH}
```

Reply:
Alright, I guess someone could still write the following to be able to override the configuration:

```javascript
environmentVariables: {
    MY_ENVIRONMENT_VARIABLE: process.env.MY_ENVIRONMENT_VARIABLE || 'some default value'
}
```

Or perhaps `--environment-variable, -e` could be exposed as a CLI option in the future.